commonTools/build_stats/BuildStatsWrappers.cmake

```diff
@@ -40,13 +40,49 @@ function(generate_build_stats_wrappers)
       generate_build_stats_wrapper_for_lang(Fortran)
     endif()
 
+    generate_build_stats_wrapper_for_op(LD "CMAKE_LD")
+    generate_build_stats_wrapper_for_op(AR "CMAKE_AR")
+    generate_build_stats_wrapper_for_op(RANLIB "CMAKE_RANLIB")
+
     set(gather_build_status "${${PROJECT_NAME}_BINARY_DIR}/gather_build_stats.sh")
     configure_file("${BUILD_STATS_SRC_DIR}/gather_build_stats.sh"
       "${gather_build_status}" COPYONLY)
   endif()
 
 endfunction()
 
+# Generate the build stats compiler wrapper for a ar/ld/ranlib
+#
+# the only difference between this and lang, is setting the proper
+# Cmake variable name, e.g., CMAKE_LANG_COMPILER  vs CMAKE_OP
+# we can resolve this by taking in name of the variable to set
+function(generate_build_stats_wrapper_for_op op_name variable_to_set)
+
+  string(TOLOWER "${op_name}" op_lc)
+  set(compiler_wrapper
+    "${${PROJECT_NAME}_BINARY_DIR}/build_stat_${op_lc}_wrapper.sh")
+
+  # Override the compiler with the wrapper but remember the original compiler
+  if ("${${variable_to_set}_ORIG}" STREQUAL "")
+    set(${variable_to_set}_ORIG "${op_lc}"
+      CACHE FILEPATH "Original non-wrappeed ${op_name}" FORCE )
+    set(${variable_to_set} "${compiler_wrapper}"
+      CACHE FILEPATH "Overwritten build stats ${op_name} compiler wrapper" FORCE )
+  endif()
+
+  message("-- " "Generate build stats compiler wrapper for ${op_name}")
+  set(BUILD_STAT_COMPILER_WRAPPER_INNER_COMPILER "${${variable_to_set}_ORIG}")
+  configure_file("${BUILD_STATS_SRC_DIR}/build_stat_lang_wrapper.sh.in"
+    "${compiler_wrapper}" @ONLY)
+  print_var(${variable_to_set})
+
+  # Use the orginal compiler for the installed <XXX>Config.cmake files
+  # doubt this works w/AR/LD/RANLIB
+  set(${variable_to_set}_COMPILER_FOR_CONFIG_FILE_INSTALL_DIR
+    "${${variable_to_set}_ORIG}" CACHE INTERNAL "")
+  print_var(${variable_to_set}_COMPILER_FOR_CONFIG_FILE_INSTALL_DIR)
+
+endfunction()
```

Comment:
I left it so if things break, it will be clear there is something screwed up in this.  Too many things tend to break w.r.t. python. Just updating Ubuntu (minor release) caused some Python problems.

> > Ross, I can make python found - the problem is there seems to be ambiguity as to whether 1) python will be guaranteed in the future, and 2) whether python will point to python2 or python3. Ubuntu seems to be saying no python and you should use python3.
> 
> As long as we are using RHEL7 at SNL we will always have a 'python' in the path. If we write our Python code to work with Python 2.7 or 3.x, we should be good to go (as long as the envs are not setting `PYTHONHOME` which seems to be screwing things up at least with the SEMS Python 3 modules.
> 
> For now, let's just do what works on SNL systems and the people can hack their env to get it working on other systems (like creating a symlink `python` pointing to `$(which python3)`). [Here](https://developers.redhat.com/blog/2018/11/14/python-in-rhel-8/#:~:text=In%20RHEL%208%2C%20Python%203.6,generally%20have%20the%20python3%20prefix.) is an interesting conversation about what `python` should point to (it points to nothing). What a mess.
> 
> Perhaps the best thing to do to make the build wrappers explicitly call Python as:
> 
> ```
>   "@PYTHON_EXECUTABLE@" \
>     "@BUILD_STATS_SRC_DIR@/wrapper/magic_wrapper.py" \
>     "----base-build-dir=${base_build_dir}" "----op=${cmd}" "$@"
> ```
> 
> instead of as:
> 
> ```
>   "@BUILD_STATS_SRC_DIR@/wrapper/magic_wrapper.py" \
>     "----base-build-dir=${base_build_dir}" "----op=${cmd}" "$@"
> ```
> 
> in [build_stat_lang_wrapper.sh.in](https://github.com/trilinos/Trilinos/blob/fca3d227ac7c83f29ba1be1f1748549fb82ce135/commonTools/build_stats/build_stat_lang_wrapper.sh.in)? I think that solves the problem. (No one should ever call `magic_wrapper.py` manually.)

The problem is I rarely use SNL SEMS. I originally wrote this stuff on ATS1 and later ATS2 platforms (that are hosted off site).

In particular, I'd like to have this deployed on the next platform that isn't to be named. If this could work robustly, then they could give us reasonable build stats before that machine hits production (and we can file issues with them to get their toolchain faster - e.g., the Kitware/Ninja/Fortran we opened  months ago).